sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/ShufflePartitionsUtil.scala

```diff
@@ -172,7 +172,7 @@ object ShufflePartitionsUtil extends Logging {
     // coalesce any partitions after the end of last skew section.
     if (numPartitions > start) {
       val partitionSpecs = coalescePartitions(
-        partitionIndices(start), partitionIndices.last + 1, validMetrics, targetSize)
+        partitionIndices(start), partitionIndices.last + 1, validMetrics, targetSize, true)
       newPartitionSpecsSeq.zip(partitionSpecs).foreach(spec => spec._1 ++= spec._2)
     }
```

Comment:
How do we make sure we return at least one partition spec if there is no skew partition spec?

Reply:
this code is under `coalescePartitionsWithSkew` which must have some skew partition specs